File: MoonEngine/Base/src/main/java/com/greentree/engine/moon/base/assets/BaseAssetSerializatorModule.java
package com.greentree.engine.moon.base.assets;

import com.greentree.engine.moon.base.AssetManagerProperty;
import com.greentree.engine.moon.base.assets.json.JSONAssetSerializator;
import com.greentree.engine.moon.base.assets.number.*;
import com.greentree.engine.moon.base.assets.properties.PropertiesAssetSerializator;
import com.greentree.engine.moon.base.assets.scene.XMLSceneAssetSerializator;
import com.greentree.engine.moon.base.assets.text.PropertyStringAssetSerializator;
import com.greentree.engine.moon.base.assets.text.ResourceToTextSerializator;
import com.greentree.engine.moon.base.assets.xml.XMLAssetSerializator;
import com.greentree.engine.moon.base.property.modules.WriteProperty;
import com.greentree.engine.moon.base.scene.SceneManagerProperty;
import com.greentree.engine.moon.modules.LaunchModule;
import com.greentree.engine.moon.modules.property.EngineProperties;

public class BaseAssetSerializatorModule implements LaunchModule {

    @WriteProperty({AssetManagerProperty.class, SceneManagerProperty.class})
    @Override
    public void launch(EngineProperties context) {
        final var manager = context.get(AssetManagerProperty.class).manager();
        manager.addSerializator(new LongAssetSerializator());
        manager.addSerializator(new IntAssetSerializator());
        manager.addSerializator(new DoubleAssetSerializator());
        manager.addSerializator(new FloatAssetSerializator());
        manager.addSerializator(new ByteAssetSerializator());
        manager.addSerializator(new ShortAssetSerializator());
        manager.addSerializator(new BooleanAssetSerializator());
        manager.addSerializator(new ResourceToTextSerializator());
        manager.addSerializator(new PropertyStringAssetSerializator());
        manager.addSerializator(new PropertiesAssetSerializator());
        manager.addSerializator(new XMLAssetSerializator());
        manager.addSerializator(new JSONAssetSerializator());
        manager.addSerializator(new XMLSceneAssetSerializator());
    }

}

File: MoonEngine/Base/src/main/java/com/greentree/engine/moon/base/assets/BaseAssetSerializatorModule.kt
package com.greentree.engine.moon.base.assets

import com.greentree.engine.moon.assets.serializator.addSerializator
import com.greentree.engine.moon.base.AssetManagerProperty
import com.greentree.engine.moon.base.assets.number.BooleanAssetSerializator
import com.greentree.engine.moon.base.assets.number.ByteAssetSerializator
import com.greentree.engine.moon.base.assets.number.DoubleAssetSerializator
import com.greentree.engine.moon.base.assets.number.FloatAssetSerializator
import com.greentree.engine.moon.base.assets.number.IntAssetSerializator
import com.greentree.engine.moon.base.assets.number.LongAssetSerializator
import com.greentree.engine.moon.base.assets.number.ShortAssetSerializator
import com.greentree.engine.moon.base.assets.properties.PropertiesAssetSerializator
import com.greentree.engine.moon.base.assets.scene.EntityAssetSerializator
import com.greentree.engine.moon.base.assets.scene.EntityDefault
import com.greentree.engine.moon.base.assets.scene.XMLSceneAssetSerializator
import com.greentree.engine.moon.base.assets.text.PropertyStringAssetSerializator
import com.greentree.engine.moon.base.assets.text.RefStringBuilderAssetSerializator
import com.greentree.engine.moon.base.assets.text.RefStringBuilderInsertAssetSerializator
import com.greentree.engine.moon.base.assets.text.ResourceToTextSerializator
import com.greentree.engine.moon.base.assets.xml.ResourceToXMLAssetSerializator
import com.greentree.engine.moon.base.assets.xml.TextToXMLAssetSerializator
import com.greentree.engine.moon.base.assets.xml.XMLChildrenAssetSerializator
import com.greentree.engine.moon.base.property.modules.WriteProperty
import com.greentree.engine.moon.base.scene.SceneManagerProperty
import com.greentree.engine.moon.modules.LaunchModule
import com.greentree.engine.moon.modules.property.EngineProperties

class BaseAssetSerializatorModule : LaunchModule {

	@WriteProperty(AssetManagerProperty::class, SceneManagerProperty::class)
	override fun launch(context: EngineProperties) {
		val manager = context[AssetManagerProperty::class.java].manager
		manager.addDefaultLoader(EntityDefault)

		manager.addSerializator(ResourceToXMLAssetSerializator)
		manager.addSerializator(TextToXMLAssetSerializator)

		manager.addSerializator(ResourceToTextSerializator)
		manager.addSerializator(PropertyStringAssetSerializator)
		manager.addSerializator(RefStringBuilderInsertAssetSerializator)

		manager.addSerializator(PropertiesAssetSerializator)

		manager.addSerializator(XMLSceneAssetSerializator)

		manager.addSerializator(RefStringBuilderAssetSerializator)

		manager.addSerializator(XMLChildrenAssetSerializator)

		manager.addSerializator(EntityAssetSerializator)

		manager.addSerializator(LongAssetSerializator)
		manager.addSerializator(IntAssetSerializator)
		manager.addSerializator(DoubleAssetSerializator)
		manager.addSerializator(FloatAssetSerializator)
		manager.addSerializator(ByteAssetSerializator)
		manager.addSerializator(ShortAssetSerializator)
		manager.addSerializator(BooleanAssetSerializator)
	}
}
